app_dart/test/service/luci_build_service_test.dart

```diff
@@ -0,0 +1,316 @@
+
+import 'package:cocoon_service/cocoon_service.dart';
+import 'package:cocoon_service/src/model/appengine/service_account_info.dart';
+import 'package:cocoon_service/src/model/luci/buildbucket.dart';
+import 'package:cocoon_service/src/request_handling/exceptions.dart';
+import 'package:cocoon_service/src/service/luci_build_service.dart';
+import 'package:mockito/mockito.dart';
+import 'package:test/test.dart';
+
+import '../src/datastore/fake_cocoon_config.dart';
+
+void main() {
+  ServiceAccountInfo serviceAccountInfo;
+  FakeConfig config;
+  MockBuildBucketClient mockBuildBucketClient;
+  group('buildsForRepositoryAndPr', () {
+    setUp(() {
+      serviceAccountInfo = const ServiceAccountInfo(email: 'abc@abcd.com');
+      config = FakeConfig(deviceLabServiceAccountValue: serviceAccountInfo);
+      mockBuildBucketClient = MockBuildBucketClient();
+    });
+    test('emptyResponse', () async {
+      when(mockBuildBucketClient.batch(any)).thenAnswer((_) async {
+        return const BatchResponse(
+          responses: <Response>[
+            Response(
+              searchBuilds: SearchBuildsResponse(
+                builds: <Build>[],
+              ),
+            ),
+          ],
+        );
+      });
+      final LuciBuildService service =
+          LuciBuildService(config, mockBuildBucketClient, serviceAccountInfo);
+      final Map<String, Build> builders =
+          await service.buildsForRepositoryAndPr('cocoon', 1, 'abcd');
+      expect(builders.keys, isEmpty);
+    });
+
+    test('someBuilders', () async {
+      when(mockBuildBucketClient.batch(any)).thenAnswer((_) async {
+        return const BatchResponse(
+          responses: <Response>[
+            Response(
+              searchBuilds: SearchBuildsResponse(
+                builds: <Build>[
+                  Build(
+                    id: 999,
+                    builderId: BuilderId(
+                      project: 'flutter',
+                      bucket: 'prod',
+                      builder: 'Mac',
+                    ),
+                    status: Status.started,
+                  )
+                ],
+              ),
+            ),
+            Response(
+              searchBuilds: SearchBuildsResponse(
+                builds: <Build>[
+                  Build(
+                    id: 998,
+                    builderId: BuilderId(
+                      project: 'flutter',
+                      bucket: 'prod',
+                      builder: 'Linux',
+                    ),
+                    status: Status.started,
+                  )
+                ],
+              ),
+            ),
+          ],
+        );
+      });
+      final LuciBuildService service =
+          LuciBuildService(config, mockBuildBucketClient, serviceAccountInfo);
+      final Map<String, Build> builders =
+          await service.buildsForRepositoryAndPr('cocoon', 1, 'abcd');
+      expect(builders.keys.toList(), equals(<String>['Mac', 'Linux']));
```

Comment:
For more test coverage and readability, we could save the builds into e.g. `build1` and `build2`. And then `expect(builds, <String, Build>{'Mac': build1, 'Linux': build2}`.

Reply:
Done for this one. For other tests the status is changing and the instances are final we need to create them where they are used.